fnet/dial.go

```diff
@@ -4,31 +4,26 @@
 package fnet
 
 import (
-	"net"
+	"net/http"
 	"time"
 
 	"github.com/onsi/ginkgo/v2"
 	"github.com/onsi/gomega"
 )
 
-// DialAndRead dials the network address, reads the data from the established connection, and asserts it against want.
-func DialAndRead(network, address, want string, maxRetry int, retryInterval time.Duration) {
-	var (
-		conn net.Conn
-		err  error
-	)
+// HTTPGetAndAssert sends an HTTP GET request to the specified URL, asserts the response status code against want, and closes the response body.
+func HTTPGetAndAssert(url string, want int, maxRetry int, retryInterval time.Duration) {
+	var err error
 	for i := 0; i < maxRetry; i++ {
-		conn, err = net.Dial(network, address)
+		// #nosec G107 // it does not matter if url is not a constant for testing.
+		resp, err := http.Get(url)
```

Comment:
nit: The default http client does not have timeout, but we may want to set one (maybe just 3~5 seconds since the traffic is only on localhost) as a good practice. More details: https://stackoverflow.com/a/25344458/6355435

Reply:
Updated in https://github.com/runfinch/common-tests/pull/42